http_request on_response does not compile with toolchain-riscv32-esp (used on esp32-c3 boards)

### The problem

The on_response trigger for the http_request component does not compile on esp32-c3 boards.

### Which version of ESPHome has the issue?

2022.12.3

### What type of installation are you using?

pip

### Which version of Home Assistant has the issue?

N/A

### What platform are you using?

ESP32

### Board

lolin_c3_mini

### Component causing the issue

http_request

### Example YAML snippet

```yaml
# Minimal yaml reproducing the problem.
wifi:
  ssid: 'ssid'

esp32:
  # nodemcu-32s works fine
  # board: nodemcu-32s
  # but lolin_c3_mini does not
  board: lolin_c3_mini
  framework:
    type: arduino

esphome:
  name: 'on-response-test'
  on_boot:
    then:
      - http_request.send:
          method: PUT
          url: https://esphome.io
          headers:
            Content-Type: application/json
          body: Some data
          verify_ssl: false
          on_response:
            then:
              - logger.log:
                  format: "Response status: %d"
                  args:
                    - status_code

logger:

http_request:
  useragent: esphome/tagreader
  timeout: 10s
```


### Anything in the logs that might be useful for us?

```txt
src/main.cpp: In function 'void setup()':
src/main.cpp:150:79: error: no matching function for call to 'esphome::Automation<long int>::Automation(esphome::http_request::HttpRequestResponseTrigger*&)'
   automation = new Automation<int32_t>(http_request_httprequestresponsetrigger);
                                                                               ^
In file included from src/esphome/components/http_request/http_request.h:6,
                 from src/esphome.h:5,
                 from src/main.cpp:3:
src/esphome/core/automation.h:234:12: note: candidate: 'esphome::Automation<Ts>::Automation(esphome::Trigger<Ts ...>*) [with Ts = {long int}]'
   explicit Automation(Trigger<Ts...> *trigger) : trigger_(trigger) { this->trigger_->set_automation_parent(this); }
            ^~~~~~~~~~
src/esphome/core/automation.h:234:12: note:   no known conversion for argument 1 from 'esphome::http_request::HttpRequestResponseTrigger*' to 'esphome::Trigger<long int>*'
src/esphome/core/automation.h:232:32: note: candidate: 'constexpr esphome::Automation<long int>::Automation(const esphome::Automation<long int>&)'
 template<typename... Ts> class Automation {
                                ^~~~~~~~~~
src/esphome/core/automation.h:232:32: note:   no known conversion for argument 1 from 'esphome::http_request::HttpRequestResponseTrigger*' to 'const esphome::Automation<long int>&'
src/esphome/core/automation.h:232:32: note: candidate: 'constexpr esphome::Automation<long int>::Automation(esphome::Automation<long int>&&)'
src/esphome/core/automation.h:232:32: note:   no known conversion for argument 1 from 'esphome::http_request::HttpRequestResponseTrigger*' to 'esphome::Automation<long int>&&'
```


### Additional information

_No response_

--- a/esphome/components/http_request/http_request.h
+++ b/esphome/components/http_request/http_request.h
@@ -31,7 +31,10 @@ struct Header {
   const char *value;
 };
 
-class HttpRequestResponseTrigger;
+class HttpRequestResponseTrigger : public Trigger<int32_t> {
+ public:
+  void process(int32_t status_code) { this->trigger(status_code); }
+};
 
 class HttpRequestComponent : public Component {
  public:
@@ -136,11 +139,6 @@ template<typename... Ts> class HttpRequestSendAction : public Action<Ts...> {
   std::map<const char *, TemplatableValue<std::string, Ts...>> json_{};
   std::function<void(Ts..., JsonObject)> json_func_{nullptr};
   std::vector<HttpRequestResponseTrigger *> response_triggers_;
-};
-
-class HttpRequestResponseTrigger : public Trigger<int> {
- public:
-  void process(int status_code) { this->trigger(status_code); }
 };
 
 }  // namespace http_request

